docs/reference/eql/functions.asciidoc

```diff
@@ -8,11 +8,60 @@ experimental::[]
 
 {es} supports the following EQL functions:
 
+* <<eql-fn-cidrmatch>>
 * <<eql-fn-endswith>>
 * <<eql-fn-length>>
 * <<eql-fn-startswith>>
 * <<eql-fn-substring>>
 
+[discrete]
+[[eql-fn-cidrmatch]]
+==== `cidrMatch`
+
+Returns `true` if an IP address is contained in one or more provided
+https://en.wikipedia.org/wiki/Classless_Inter-Domain_Routing[CIDR] blocks.
+
+[%collapsible]
+====
+*Example*
+
+[source,eql]
+----
+// source.address = "192.168.152.12"
+cidrMatch(source.address, "192.168.0.0/16")               // returns true
+cidrMatch(source.address, "192.168.0.0/16", "10.0.0.0/8") // returns true
+cidrMatch(source.address, "10.0.0.0/8")                   // returns false
+cidrMatch(source.address, "10.0.0.0/8", "10.128.0.0/9")   // returns false
```

Comment:
@aleksmaus When I tested this using the EQL search API, it didn't return results as expected. I included duplication steps below in case I made an error somewhere.

Create an index mapping to ensure `source.address` is an IP field.
```
PUT /my_index
{
  "mappings": {
    "properties": {
      "source": {
        "properties": {
          "address": {
            "type": "ip"
          }
        }
      }
    }
  }
}
```

Index "192.168.152.12" as a `source.address` value.
```
PUT /my_index/_doc/1
{
  "@timestamp": "2020-12-06T11:04:05.000Z",
  "event": {
    "category": "process"
  },
  "source": {
    "address": "192.168.152.12"
  }
}
```

Use `cidrMatch` to search for "192.168.152.12" in the "192.168.0.0/16" CIDR block. This should be `true` and return _doc `1` as a result.
```
{
GET /my_index/_eql/search
  "query": "process where cidrMatch(source.address, \"192.168.0.0/16\") == true "
}
```

Instead, I get no results.
```
{
  "took" : 2,
  "timed_out" : false,
  "hits" : {
    "total" : {
      "value" : 0,
      "relation" : "eq"
    },
    "events" : [ ]
  }
}
```

If I change `true` to `false`, the _doc is returned.
```
{
GET /my_index/_eql/search
  "query": "process where cidrMatch(source.address, \"192.168.0.0/16\") == true "
}
```

Results:

```
{
  "took" : 2,
  "timed_out" : false,
  "hits" : {
    "total" : {
      "value" : 1,
      "relation" : "eq"
    },
    "events" : [
      {
        "_index" : "my_index",
        "_id" : "1",
        "_score" : null,
        "_source" : {
          "@timestamp" : "2020-12-06T11:04:05.000Z",
          "event" : {
            "category" : "process"
          },
          "source" : {
            "address" : "192.168.152.12"
          }
        },
        "sort" : [
          1607252645000
        ]
      }
    ]
  }
}
```

Reply:
I created https://github.com/elastic/elasticsearch/issues/55709 to track this bug.